celery_utils/logged_task.py

```diff
@@ -36,6 +36,14 @@ def on_retry(self, exc, task_id, args, kwargs, einfo):
         """
         Capture the exception that caused the task to be retried, if any.
         """
-        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
         if exc is not None:
             log.warning('[{}] retried due to {}'.format(task_id, einfo.traceback))
+        super(LoggedTask, self).on_retry(exc, task_id, args, kwargs, einfo)
+
+    def on_failure(self, exc, task_id, args, kwargs, einfo):
```

Comment:
Looking at their code in more depth, I also wonder if we couldn't just configure our logging to emit >= ERROR logs from celery itself. It looks like they have pretty extensive logging built in:

retry: https://github.com/celery/celery/blob/fd8b52e3c8ab25652f56198b52bdcb60d7de65b2/celery/app/trace.py#L187
error: https://github.com/celery/celery/blob/fd8b52e3c8ab25652f56198b52bdcb60d7de65b2/celery/app/trace.py#L223